designs/cache_options.md

```diff
@@ -0,0 +1,104 @@
+Cache Options
+===================
+
+This document describes how we imagine the cache options to look like in
+the future.
+
+## Goals
+
+* Align everyone on what settings on the cache we want to support and
+  their configuration surface
+* Ensure that we support both complicated cache setups and provide an
+  intuitive configuration UX
+
+## Non-Goals
+
+* Line out how the actual implementation of the cache itself will look like.
+  The assumption is that the most granular level we will end up with is
+  "per object multiple namespaces with distinct selectors" and that this
+  can be implemented using a "meta cache" that delegates per object and by
+  extending the current multi-namespace cache
+* Outline any kind of timeline for when these settings will be implemented.
+  Implementation will happen gradually over time whenever someone steps up
+  to do the actual work
+
+## Proposal
+
+
+```
+type CacheSetting struct {
```

Comment:
It would be super helpful to have implementation examples (with `CacheSetting`) of different scenarios in the proposal.